src/symbolize/gimli.rs

```diff
@@ -217,8 +220,24 @@ cfg_if::cfg_if! {
                 if !elf.little_endian {
                     return None;
                 }
-                let mut syms = elf.syms.iter().collect::<Vec<_>>();
-                syms.sort_unstable_by_key(|s| s.st_value);
+                let mut syms = elf
+                    .syms
+                    .iter()
+                    .map(|s| (s, false))
+                    .chain(elf.dynsyms.iter().map(|s| (s, true)))
+                    // Only look at function/object symbols. Not entirely sure
+                    // why, but this is what libbacktrace does.
```

Comment:
I guess the assumption is that only functions contain executable code.

Reply:
The reason I ended up putting this in is that without it a few addresses were still falling back to `dladdr`. That meant that there's some symbol here which was around the same address as a function address and ended up getting mistaken as a function address.

Do you know what might be causing that? Are the relative `st_value` fields for symbols relative to their own sections, so we only care about symbols relative to the text section?

Reply:
I don't know what might cause that. In executables and shared objects, `st_value` is a virtual address. I think the exception to that is TLS symbols?